ocaml/xenopsd/xc/service.mli

```diff
@@ -0,0 +1,113 @@
+(* Copyright (C) Citrix Systems Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+*)
+
+module Qemu : sig
+  module SignalMask : sig
+    type t
+
+    val create : unit -> t
+
+    val set : t -> int -> unit
+
+    val unset : t -> int -> unit
+
+    val has : t -> int -> bool
+  end
+
+  val signal_mask : SignalMask.t
+
+  val name : string
+
+  val pidxenstore_path_signal : Xenctrl.domid -> string
+
+  val pidfile_path : Xenctrl.domid -> string
+  (** path of file containing the pid value *)
+
+  val start_daemon :
+       path:string
+    -> args:string list
+    -> domid:Xenctrl.domid
+    -> ?fds:(string * Unix.file_descr) list
+    -> unit
+    -> Forkhelpers.pidty
+
+  val pid : xs:Xenstore.Xs.xsh -> Xenctrl.domid -> int option
+
+  val is_running : xs:Xenstore.Xs.xsh -> Xenctrl.domid -> bool
+
+  val stop :
+    xs:Xenstore.Xs.xsh -> qemu_domid:Xenctrl.domid -> Xenctrl.domid -> unit
+end
+
+module Vgpu : sig
+  val start :
+       xs:Xenstore.Xs.xsh
+    -> vcpus:int
+    -> vgpus:Xenops_interface.Vgpu.t list
+    -> restore:bool
+    -> Xenops_task.Xenops_task.task_handle
+    -> int
+    -> unit
+
+  val cancel_key : Xenctrl.domid -> Cancel_utils.key
+
+  val state_path : Xenctrl.domid -> string
+
+  val pid : xs:Xenstore.Xs.xsh -> Xenctrl.domid -> int option
```

Comment:
Does a `None` result imply that `is_running` is false? This would suggest that `is_running` is a simple test for `pid <> None`.

Reply:
Not exactly, is_running is:
```
  let is_running ~xs domid =
    match pid ~xs domid with
    | None ->
        false
    | Some p -> (
      try Unix.kill p 0 ; (* This checks the existence of pid p *)
                          true
      with _ -> false
    )
```